### PR TITLE
End Reasons implementation for Contracts

### PR DESCRIPTION
**Pull request by JoaFS**
This PR have the implementation of the End Reasons for Contracts.

New functions:
- Manage end reasons (Create, Read, Update and Delete) through the "End Reasons" section in "Settings" menu.
- Add an end reason to a contract by editing it through the Contracts menu.
- Delete an end reason to a contract by editing it through the Contracts menu.

### CODE REVIEW

File: src/types/contracts.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ export interface IContract {
   notes: string;
   contractType: string;
   paymentSettlement: IPaymentSettlement[];
+  endReasonID: string;
```
Comment (maxituc21) [inline src/types/contracts.ts:15]: Please change to `endReasonId `to keep to the pattern.

File: src/endReasons.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  </List>
+);
+
+export const EndReasonsEdit = () => (
+  <EditForm formData={formData} title="End Reasons" resource="end-reasons" />
+);
+
+export const EndReasonsCreate = () => (
+  <CreateForm formData={formData} title="End Reasons" resource="end-reasons" />
+);
```
Comment (maxituc21) [inline src/endReasons.js:28]: Please add a blank line at the end of the file to fix this warning.

File: src/types/contracts.ts (hunk 1 above)
Comment (JoaFS) [reply src/types/contracts.ts]: Changed.

File: src/endReasons.js (hunk 2 above)
Comment (JoaFS) [reply src/endReasons.js]: Added.

**Review by maxituc21 — APPROVED** [review]

**Comment by eazaretzky** [comment]
nice work @JoaFS, could you please add a few screenshot of how the change looks in the UI?

**Review by eazaretzky — APPROVED** [review]

**Comment by JoaFS** [comment]
> nice work @JoaFS, could you please add a few screenshot of how the change looks in the UI?

Of course! Here are some screenshots of how it looks with the new changes in the UI
image
image
image
image

**Pull request merged**